pkg/finality-grandpa/bridge_state.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"sync"
+)
+
+type waker struct {
+	mtx    sync.RWMutex
+	wakeCh chan any
+}
+
+func newWaker() *waker {
+	return &waker{wakeCh: make(chan any, 1000)}
+}
+
+func (w *waker) wake() {
+	w.mtx.RLock()
+	defer w.mtx.RUnlock()
+	if w.wakeCh == nil {
+		return
+	}
+	go func() {
+		select {
+		case w.wakeCh <- nil:
+		default:
+		}
+	}()
```

Comment:
This is either a clever piece of code or a mistake. But I don't understand what it does. Can you please tell?